tests/test_posixDatastoreFits.py

```diff
@@ -0,0 +1,178 @@
+        self.assertCatalogEqual(catalog, catalogOut)
+        # Remove
+        datastore.remove(uri)
+        # Get should now fail
+        with self.assertRaises(ValueError):
+            datastore.get(uri, storageClass=storageClass, parameters=None)
+        # Can only delete once
+        with self.assertRaises(FileNotFoundError):
+            datastore.remove(uri)
+
+    def testTransfer(self):
+        catalog = self.makeExampleCatalog()
+        path = "tester.fits"
+        inputConfig = DatastoreConfig(self.configFile)
+        inputConfig['datastore.root'] = os.path.join(self.testDir, "./test_input_datastore")
+        inputPosixDatastore = PosixDatastore(config=inputConfig)
+        outputConfig = inputConfig.copy()
+        outputConfig['datastore.root'] = os.path.join(self.testDir, "./test_output_datastore")
+        outputPosixDatastore = PosixDatastore(config=outputConfig)
+        storageClass = outputPosixDatastore.storageClassFactory.getStorageClass("SourceCatalog")
+        inputUri, _ = inputPosixDatastore.put(catalog, storageClass, path)
+        outputUri, _ = outputPosixDatastore.transfer(inputPosixDatastore, inputUri, storageClass, path)
+        catalogOut = outputPosixDatastore.get(outputUri, storageClass)
+        self.assertCatalogEqual(catalog, catalogOut)
+
+
+class PosixDatastoreExposureTestCase(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        if lsst.afw.image is None:
+            raise unittest.SkipTest("afw not available.")
+
+    def setUp(self):
+        self.testDir = os.path.dirname(__file__)
+        self.configFile = os.path.join(self.testDir, "config/basic/butler.yaml")
+
+    def testExposurePutGet(self):
+        example = os.path.join(self.testDir, "data", "basic", "small.fits")
+        exposure = lsst.afw.image.ExposureF(example)
+        datastore = PosixDatastore(config=self.configFile)
+        # Put
+        storageClass = datastore.storageClassFactory.getStorageClass("ExposureF")
+        uri, comps = datastore.put(exposure, storageClass=storageClass, storageHint="test_exposure.fits",
+                                   typeName=None)
+        # Get
+        exposureOut = datastore.get(uri, storageClass=storageClass, parameters=None)
+        self.assertEqual(type(exposure), type(exposureOut))
+
+        # Get some components
+        for c in ("wcs", "image", "mask", "coaddInputs", "psf"):
+            self.assertIn(c, comps)
+            component = datastore.get(comps[c], storageClass=storageClass)
+            self.assertIsNotNone(component)
+
+        # Get a component to check it
+        self.assertIn("wcs", comps)
+        wcs = datastore.get(comps["wcs"], storageClass=storageClass)
+
+        # Simple check of WCS
+        bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(0, 0),
+                                   lsst.afw.geom.Extent2I(9, 9))
+        self.assertWcsAlmostEqualOverBBox(wcs, exposure.getWcs(), bbox)
```

Comment:
Test the actual component values?

Reply:
This is testing the Wcs values. I don't necessarily know how to test every single Exposure component but I can default to seeing if `==` works.